core/src/main/scala-3/pureconfig/generic/derivation/ConfigReaderDerivation.scala

```diff
@@ -0,0 +1,47 @@
+package pureconfig
+package generic
+package derivation
+
+import scala.deriving.Mirror
+
+import pureconfig.generic.{CoproductHint, ProductHint}
+
+class ConfigReaderDerivation(coproductHint: CoproductHint[?], productHint: ProductHint[?])
+    extends CoproductConfigReaderDerivation(coproductHint)
+    with ProductConfigReaderDerivation(productHint) {
+
+  trait DerivedConfigReader[A] extends ConfigReader[A]
+  object DerivedConfigReader {
+    inline def derived[A](using m: Mirror.Of[A]): DerivedConfigReader[A] =
+      inline m match {
+        case given Mirror.ProductOf[A] => ProductConfigReader.derived
+        case given Mirror.SumOf[A] => CoproductConfigReader.derived
+      }
+  }
+
+  def withCoproductHint(coproductHint: CoproductHint[?]) =
+    new ConfigReaderDerivation(
+      coproductHint,
+      productHint
+    )
+
+  def withProductHint(productHint: ProductHint[?]) =
+    new ConfigReaderDerivation(
+      coproductHint,
+      productHint
+    )
+
+  extension (c: ConfigReader.type)
+    inline def derived[A](using m: Mirror.Of[A]): ConfigReader[A] =
+      DerivedConfigReader.derived
+}
+
+object ConfigReaderDerivation {
+  object Default
+      extends ConfigReaderDerivation(
+        CoproductHint.default[Any],
+        ProductHint.default[Any]
+      )
+}
+
+val default = ConfigReaderDerivation.Default
```

Comment:
So it's preferable that I remove the definition and that the user uses `import pureconfig.generic.derivation.ConfigReaderDerivation.Default._` instead?

But I'd just like to point out that if the user is already writing Scala 3 code, they'd know that top level definitions might not be a package. Also, Scala 2 already had package objects that did the same thing (I would have put this definition in a package object in Scala 2). Also not sure why a user would care if it's a package or an object, as long as they get their definitions. So from a user perspective this shouldn't be a problem. Just curious to why you think this might cause any issues, if I am missing something. :)

Reply:
Fair point. The organization still feels a bit awkward to me, mostly because the intended usage is to import a value while I expect (without a strong reason) package structures to be static. But you're right that it's unrelated to Scala 2 vs Scala 3 and it's a personal choice, so unless other maintainers share this preference please let this as it is.